beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -1045,33 +1045,51 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
 
     /// Returns the blobs at the given root, if any.
     ///
-    /// ## Errors
+    /// Returns `Ok(None)` if the blobs are not found. This could indicate the blob has been pruned
+    /// or that the block it is referenced by doesn't exist in our database.
     ///
-    /// May return a database error.
+    /// If we can find the corresponding block in our database, we know whether we *should* have
+    /// blobs. If we should have blobs and no blobs are found, this will error. If we shouldn't,
+    /// this will reconstruct an empty `BlobsSidecar`.
+    ///
+    /// ## Errors
+    /// - any database read errors
+    /// - block and blobs are inconsistent in the database
+    /// - this method is called with a pre-eip4844 block root
     pub fn get_blobs(
```

Comment:
The re-work here was mostly to make sure this `chain.get_blobs` returns: 

1. an error when this is called for a pre-eip4844 block root
2. an error when we don't have blobs that we should have
3. `Ok(None)` when blobs has been pruned